[SuhailLoya] iP

SuhailLoya opened the pull request:
# Lumina

> Meet Lumina: your personal task management assistant. Free your mind from all the organizing, and focus on accomplishing what truly matters.

It's 
- Text based
- Easy-to-learn
- ~~FAST~~ SUPER FAST to use

All you need to do is
1. Clone this repo [here](https://github.com/SuhailLoya/ip)
2. Build and run
3. Enjoy :smile:

And, it's **FREE!**

Features:
- [x] Managing tasks
- [x] Managing deadlines
- [ ] Reminders (coming soon)

#
If you are a Java programmer, you can use to practice Jave too. Here's the `main` method:

```java
public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        Lumina lumina = new Lumina();
        lumina.echo(sc);
    }
```
> [!TIP]
> To maintain clean and manageable code, always _follow coding standards._

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

SuhailLoya committed:
Duke.java: Rename to Lumina.java; implement basic skeleton

SuhailLoya committed:
Add echo and indentMessage functions

SuhailLoya committed:
add list functionality

SuhailLoya committed:
add mark / unmark task ability

SuhailLoya committed:
add different types of tasks

SuhailLoya committed:
setup text-ui-test

SuhailLoya committed:
add error handling

SuhailLoya committed:
add error handling

SuhailLoya committed:
Add delete functionality

SuhailLoya committed:
refactor: Prevent instantiation of more than 1 Lumina instance

SuhailLoya committed:
Add createDataDirectoryAndFileIfNotExists Method

SuhailLoya committed:
Add support for reading data

SuhailLoya committed:
Add comments for read and load data methods

SuhailLoya committed:
fix(Task.java): change access permissions for description and isDone to private

SuhailLoya committed:
Add support for saving data when exiting

SuhailLoya committed:
Merge branch 'branch-Level-7'

SuhailLoya committed:
Add sample data file

SuhailLoya committed:
Add support for date and time

SuhailLoya committed:
Merge branch 'branch-Level-8'

SuhailLoya committed:
Add more classes to achieve more oop-like practices

SuhailLoya committed:
Lumina.java: clean up

SuhailLoya committed:
add packages

SuhailLoya committed:
Merge branch 'add-gradle-support'

SuhailLoya committed:
add tests

SuhailLoya committed:
Remove references to duke from build.gradle

SuhailLoya committed:
add find functionality

SuhailLoya committed:
fix: coding style

SuhailLoya committed:
add javadocs

SuhailLoya committed:
Merge branch 'branch-A-JavaDoc'

SuhailLoya committed:
Merge branch 'branch-A-CodingStandard'

SuhailLoya committed:
Resolve merge conflicts

SuhailLoya committed:
fix bug after merge conflict

SuhailLoya committed:
fix: rename reference to lumina.main.Lumina in java doc to Lumina

adriantan101 commented on src/main/java/lumina/main/Lumina.java:
```diff
@@ -0,0 +1,183 @@
+    private static int luminaCount = 0;
+
+    // composed objects
+    private Ui ui;
+    private Parser parser;
+    private Storage storage;
+    private TaskList taskList;
```
I like the naming here! They are short and descriptive enough to explain their roles in the application 👍

adriantan101 commented on src/main/java/lumina/main/Lumina.java:
```diff
@@ -0,0 +1,183 @@
+                    String[] msgSplit = msg.split(" ");
+                    if (msgSplit.length == 2) {
+                        int taskIndex = Integer.parseInt(msgSplit[1]) - 1; // 0 indexed
+                        taskList.markTaskNotDone(taskIndex);
```
I like the naming of markTaskDone, markTaskNotDone, deleteTask, listTasks, findTasks as methods as using verbs to describe the actions performed on the tasks which is consistent with the naming conventions.

adriantan101 commented on src/main/java/lumina/main/Lumina.java:
```diff
@@ -0,0 +1,183 @@
+                try {
+                    String[] msgSplit = msg.split(" ");
+                    if (msgSplit.length == 2) {
+                        int taskIndex = Integer.parseInt(msgSplit[1]) - 1; // 0 indexed
```
While msgSplit is understandable, perhaps renaming it to splitMessage or commandParts would make the code slightly more readable?

adriantan101 commented on src/main/java/lumina/parser/Parser.java:
```diff
@@ -0,0 +1,94 @@
+ */
+public class Parser {
+
+    /**
+     * Parses a date string into a LocalDate object.
+     *
+     * @param date the date string to parse
+     * @return the parsed LocalDate object
+     * @throws LuminaException if the date string is in an invalid format
+     */
```
JavaDoc comments are descriptive of the class/methods 👍

adriantan101 commented on src/main/java/lumina/parser/Parser.java:
```diff
@@ -0,0 +1,94 @@
+     * @throws LuminaException if the date string is in an invalid format
+     */
+    public LocalDate parseDateString(String date) throws LuminaException {
+        LocalDate ret = null;
```
Maybe a better name would be parsedDate or dateObject?

adriantan101 reviewed (commented):
LGTM! Other than the small nits around naming of variables, code quality and style are mostly adhered to 🫡

chashaobao123 commented on src/main/java/lumina/parser/Parser.java:
```diff
@@ -0,0 +1,94 @@
+                    if (parts.length != 5) {
+                        throw luminaException;
+                    }
+                    String startDateTime = parts[3].trim();
```
I like the naming of startDateTime and endDateTime (as opposed to from and to) here! they are very clear.

chashaobao123 commented on src/main/java/lumina/storage/Storage.java:
```diff
@@ -0,0 +1,112 @@
+     */
+    public void saveData(TaskList taskList) {
+        Path path = Paths.get(DATA_FILE_PATH);
+        ArrayList<String> lines = taskList.toLines();
```
Maybe changing the variable names from lines to taskData would be clearer?

chashaobao123 commented on src/main/java/lumina/task/TaskList.java:
```diff
@@ -0,0 +1,332 @@
+                listedTaskMessage.append("\n");
+            }
+        }
+        ui.printMessage(listedTaskMessage.toString());
```
Instead of passing the message, would passing the tasks ArrayList to Ui to print be a better abstraction?

chashaobao123 commented on src/main/java/lumina/task/TodoTask.java:
```diff
@@ -0,0 +1,55 @@
+package lumina.task;
+
+/**
```
I like how you included this comment! It makes it extra clear that the code isn't missing anything

chashaobao123 commented on src/main/java/lumina/storage/Storage.java:
```diff
@@ -0,0 +1,112 @@
+    }
+
+    private ArrayList<Task> readData() {
+        // intialize empty array
```
I like how there is a comment on almost every line to detail exactly what the line does. this makes it very clear to read, especially for beginners

SuhailLoya committed:
add config files for checkstyle

SuhailLoya committed:
fix checkstyle issues

SuhailLoya committed:
Merge branch 'branch-A-CheckStyle'

SuhailLoya committed:
Add basic gui

SuhailLoya committed:
Merge branch 'branch-Level-10'

SuhailLoya committed:
Fix javadocs for clarity

SuhailLoya committed:
Modify build.gradle to specify archiveFileName

SuhailLoya committed:
Add assert statements

SuhailLoya committed:
Fix code quality at different locations in the code

SuhailLoya committed:
Merge pull request #2 from SuhailLoya/branch-A-CodeQuality

Fix code quality at different locations in the code

SuhailLoya committed:
Resolve merge conflicts

SuhailLoya committed:
Storage.jave: Add assert for file exists

The readData() function in Storage.java assumes the data file already exists, which could lead to errors if the file is missing.

To prevent these errors, this commit adds an assertion to check for
file's existence before attempting to read the file.

This approach ensures that the function's assumptions are validated
in runtime.

SuhailLoya committed:
Merge pull request #3 from SuhailLoya/branch-A-Assertions

Add assert statements checks to validate assumptions

SuhailLoya committed:
fix remind task

SuhailLoya committed:
Remove duke references and add Ui.png for docs

SuhailLoya committed:
Edit Ui.png

SuhailLoya committed:
Add user guide

SuhailLoya committed:
Edit user guide

SuhailLoya committed:
Add print statements for creating files and directories

SuhailLoya committed:
Remove unnecessary line

SuhailLoya committed:
Fix typo in user guide
